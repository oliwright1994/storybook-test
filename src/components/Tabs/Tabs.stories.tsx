import createStory from '@bright/storybook/createStory'
import React from 'react'

import Tabs from '.'

createStory('COMPONENTS | Tabs', [
  {
    label: 'Tabs',
    jsx: () => {
      return (
        <Tabs>
          <Tabs.TabList>
            <Tabs.Tab>Tab1</Tabs.Tab>
            <Tabs.Tab>Tab2</Tabs.Tab>
            <Tabs.Tab>Tab2</Tabs.Tab>
          </Tabs.TabList>
          <Tabs.TabPanels>
            <Tabs.TabPanel>Panel One</Tabs.TabPanel>
            <Tabs.TabPanel>Panel Two</Tabs.TabPanel>
            <Tabs.TabPanel>Panel Three</Tabs.TabPanel>
          </Tabs.TabPanels>
        </Tabs>
      )
    },
  },
])
