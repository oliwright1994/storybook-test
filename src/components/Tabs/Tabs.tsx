import { ITabs as IChakraTabs, Tabs as ChakraTabs } from '@chakra-ui/core'
import React, { FC } from 'react'

import Tab from './Tab'
import TabList from './TabList'
import TabPanel from './TabPanel'
import TabPanels from './TabPanels'

interface ITabs extends FC<IChakraTabs> {
  Tab: typeof Tab
  TabList: typeof TabList
  TabPanel: typeof TabPanel
  TabPanels: typeof TabPanels
}

const Tabs: ITabs = ({ children }) => {
  return <ChakraTabs  flexGrow={1}>{children}</ChakraTabs>
}

Tabs.Tab = Tab
Tabs.TabList = TabList
Tabs.TabPanel = TabPanel
Tabs.TabPanels = TabPanels

export default Tabs
