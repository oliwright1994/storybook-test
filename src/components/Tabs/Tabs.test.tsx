import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Tabs from '.'

describe('<Tabs />', () => {
  const labels = {
    tab: 'Tab',
    panel: 'Panel',
  }
  const tabsArray = ['0', '1', '2']
  const getLabel = (id: string, type: string): string => `${type}${id}`
  const tabs = tabsArray.map(id => <Tabs.Tab key={id}>{getLabel(id, labels.tab)}</Tabs.Tab>)
  const panels = tabsArray.map(id => <Tabs.TabPanel key={id}>{getLabel(id, labels.panel)}</Tabs.TabPanel>)

  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(
      <Tabs>
        <Tabs.TabList>{tabs}</Tabs.TabList>
        <Tabs.TabPanels>{panels}</Tabs.TabPanels>
      </Tabs>
    )

    expect(container).toMatchSnapshot()
  })
})
