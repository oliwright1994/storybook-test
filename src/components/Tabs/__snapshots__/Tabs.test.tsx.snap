// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render as expected 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      aria-orientation="horizontal"
      class="css-1dxxh68"
      role="tablist"
    >
      <button
        aria-controls="panel:1-0"
        aria-selected="true"
        class="css-1kt89oh"
        id="tab:1-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        Tab0
      </button>
      <button
        aria-controls="panel:1-1"
        aria-selected="false"
        class="css-1kt89oh"
        id="tab:1-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab1
      </button>
      <button
        aria-controls="panel:1-2"
        aria-selected="false"
        class="css-1kt89oh"
        id="tab:1-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab2
      </button>
    </div>
    <div
      class="css-0"
      tabindex="-1"
    >
      <div
        aria-labelledby="tab:undefined"
        class="css-1jj07fx"
        id="panel:undefined"
        role="tabpanel"
        tabindex="-1"
      >
        Panel0
      </div>
      <div
        aria-labelledby="tab:undefined"
        class="css-1jj07fx"
        hidden=""
        id="panel:undefined"
        role="tabpanel"
        tabindex="-1"
      >
        Panel1
      </div>
      <div
        aria-labelledby="tab:undefined"
        class="css-1jj07fx"
        hidden=""
        id="panel:undefined"
        role="tabpanel"
        tabindex="-1"
      >
        Panel2
      </div>
    </div>
  </div>
</div>
`;
